Coupons: Update button titles on Coupon Restrictions screen to reflect the number of selected items

itsmeichigo opened the pull request:
<!-- Remember about a good descriptive title. -->

Closes: #6491 
<!-- Id number of the GitHub issue this PR addresses. -->

### Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->
This PR wraps up the Coupon Restrictions screen by updating the titles and icons of Exclude Products and Exclude Product Categories buttons.

This was handled by updating the view model with logic to check for selected items to return the appropriate icons and titles.

### Testing instructions
<!-- Step by step testing instructions. When necessary break out individual scenarios that need testing, consider including a checklist for the reviewer to go through. -->
- Navigate to Menu > Coupons > select a coupon in your store.
- Select the ellipsis button on the top right and select Edit Coupon.
- On the Edit Coupon screen select Usage Restrictions.
- Select the Exclude Products and Exclude Product Categories button, select and deselect any items to see if the buttons are updated correctly on the Usage Restrictions screen.

### Screenshots
<!-- Include before and after images or gifs when appropriate. -->

https://user-images.githubusercontent.com/5533851/167060522-83c2268d-fe9a-4641-88e4-ee5d091b75ce.mp4



---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

itsmeichigo committed:
Add logic to update buttons' titles and icons to view model

itsmeichigo committed:
Update buttons' icons and titles based on the view model

itsmeichigo committed:
Update unit tests for coupon restrictions' buttons

selanthiraiyan commented on WooCommerce/Classes/ViewRelated/Coupons/Add and Edit Coupons/UsageDetails/CouponRestrictionsViewModel.swift:
```diff
@@ -113,3 +137,26 @@ final class CouponRestrictionsViewModel: ObservableObject {
+        )
+        static let excludeProductCategories = NSLocalizedString(
+            "Exclude Product Categories",
+            comment: "Title of the action button to add product categories on Coupon Usage Details screen"
```
Should this be `Title of the action button to exclude product categories on Coupon Usage Details screen`? 🤔

selanthiraiyan commented on WooCommerce/WooCommerceTests/ViewRelated/Coupons/CouponRestrictionsViewModelTests.swift:
```diff
@@ -26,4 +26,46 @@ final class CouponRestrictionsViewModelTests: XCTestCase {
+        let viewModel = CouponRestrictionsViewModel(coupon: coupon)
+
+        // Then
+        XCTAssertEqual(viewModel.excludeProductsButtonIcon.pngData(), UIImage.plusImage.pngData())
```
I didn't know that we can compare images this way. 👍 💯

selanthiraiyan commented on WooCommerce/Classes/ViewRelated/Coupons/Add and Edit Coupons/UsageDetails/CouponRestrictionsViewModel.swift:
```diff
@@ -113,3 +137,26 @@ final class CouponRestrictionsViewModel: ObservableObject {
+        static let excludedProductCategories = NSLocalizedString(
+            "Exclude Product Categories (%1$d)",
+            comment: "Title of the action button to add excluded product categories on Coupon Usage Details screen. " +
+            "The number of selected items is included between the brackets. Reads like: Exclude Product Categories (4)"
```
super nit: `The number of selected items is included between the brackets.` I would probably replace the word `included` with `mentioned`, just because `included` and `excluded` have specific meanings in this context. 😅

selanthiraiyan reviewed (approved):
Code ✅ 
- Nice work moving the logic to the view model and adding tests. ❤️ 

Testing ✅ 


:shipit:

itsmeichigo committed:
Update comments for the localized strings

itsmeichigo commented:
Thank you @selanthiraiyan for the comments! I updated the comments of the localized strings with your suggestions and also updated the screen name to match the design in 418c6bb602aefd2f21cfb3cddcbafcf525a8e76b 😅
